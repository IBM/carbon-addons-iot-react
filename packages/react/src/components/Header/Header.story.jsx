import React from 'react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import NotificationOn from '@carbon/icons-react/lib/notification/16';
import HeaderHelp from '@carbon/icons-react/lib/help/16';
import Avatar from '@carbon/icons-react/lib/user--avatar/16';

import { settings } from '../../constants/Settings';
import { Tag } from '../Tag';

import Header from './Header';

const { iotPrefix } = settings;

React.Fragment = ({ children }) => children;

const HeaderProps = {
  user: 'JohnDoe@ibm.com',
  tenant: 'TenantId: Acme',
  url: 'http://localhost:8989',
  className: 'custom-class-name',
  appName: 'Application Name',
  skipto: 'skip',
  i18n: {
    openMenu: 'Open the menu',
  },
  actionItems: [
    {
      label: 'alerts',
      onClick: action('click fired'),
      btnContent: <NotificationOn fill="white" description="Icon" />,
    },
    {
      label: 'help',
      hasHeaderPanel: true,
      btnContent: (
        <HeaderHelp
          fill="white"
          description="Icon"
          className="bx--header__menu-item bx--header__menu-title"
        />
      ),
      childContent: [
        {
          metaData: {
            href: 'http://google.com',
            title: 'this is a title',
            target: '_blank',
            rel: 'noopener noreferrer',
            element: 'a',
          },
          content: 'this is my message to you',
        },
        {
          metaData: {
            onClick: action('do another action'),
            className: 'this',
            element: 'button',
          },
          content: (
            <span>
              JohnDoe@ibm.com
              <Avatar fill="white" description="Icon" />
            </span>
          ),
        },
      ],
    },
    {
      label: 'user',
      btnContent: <Avatar fill="white" description="Icon" />,
      childContent: [
        {
          metaData: {
            href: 'http://google.com',
            title: 'this is a title',
            target: '_blank',
            rel: 'noopener noreferrer',
            element: 'a',
          },
          content: 'this is my message to you',
        },
        {
          metaData: {
            onClick: action('do another action'),
            className: 'this',
            element: 'button',
          },
          content: (
            <span>
              JohnDoe@ibm.com
              <Avatar fill="white" description="Icon" />
            </span>
          ),
        },
      ],
    },
  ],
};

const HeaderMenuProps = {
  user: 'JohnDoe@ibm.com',
  tenant: 'TenantId: Acme',
  url: 'http://localhost:8989',
  className: 'custom-class-name',
  appName: 'Application Name',
  skipto: 'skip',
  actionItems: [
    {
      label: 'user',
      btnContent: <Avatar fill="white" description="Icon" />,
      childContent: [
        {
          metaData: {
            href: 'http://google.com',
            title: 'this is a title',
            target: '_blank',
            rel: 'noopener noreferrer',
            element: 'a',
          },
          content: 'this is my message to you',
        },
        {
          metaData: {
            onClick: action('do another action'),
            className: 'this',
            element: 'button',
          },
          content: (
            <span>
              JohnDoe@ibm.com
              <Avatar fill="white" description="Icon" />
            </span>
          ),
        },
      ],
    },
  ],
};

const headerPanel = {
  className: 'header-panel',

  content: React.forwardRef((props, ref) => (
    // eslint-disable-next-line jsx-a11y/anchor-is-valid
    <a href="#" ref={ref} {...props}>
      Header panel content
    </a>
  )),
};

export default {
  title: 'Watson IoT/Header',

  parameters: {
    component: Header,
  },
};

export const HeaderActionButtonsWithDropdowns = () => (
  <div style={{ width: '100%', height: '100vh' }}>
    <Header
      {...HeaderProps}
      className={`${iotPrefix}--header--story-test-class`}
      headerPanel={headerPanel}
      appSwitcherLabel={text('AppSwitcher label', 'AppSwitcher')}
    />
    <div id="skip" />
  </div>
);

HeaderActionButtonsWithDropdowns.story = {
  name: 'Header action buttons with dropdowns',
};

export const HeaderSubmenu = () => (
  <div style={{ width: '100%', height: '100vh' }}>
    <Header {...HeaderMenuProps} className={`${iotPrefix}--header--story-test-class`} />
  </div>
);

HeaderSubmenu.story = {
  name: 'header submenu',
};

export const HeaderNoSubmenu = () => (
  <Header
    {...HeaderProps}
    className={`${iotPrefix}--header--story-test-class`}
    actionItems={[
      {
        label: 'user',
        onClick: action('click'),
        btnContent: <Avatar fill="white" description="Icon" />,
      },
    ]}
  />
);

HeaderNoSubmenu.story = {
  name: 'Header no submenu',
};

export const HeaderSubtitle = () => (
  <div style={{ width: '100%', height: '100vh' }}>
    <Header {...HeaderMenuProps} subtitle="Monitor" />
  </div>
);

HeaderSubtitle.story = {
  name: 'header subtitle',
};

export const HeaderComponentSubtitle = () => (
  <div style={{ width: '100%', height: '100vh' }}>
    <Header
      {...HeaderMenuProps}
      className={`${iotPrefix}--header--story-test-class`}
      subtitle={
        <div>
          {'Monitor'}
          <span style={{ 'margin-left': '1rem' }}>
            <Tag>Admin Mode</Tag>
          </span>
        </div>
      }
    />
  </div>
);

HeaderComponentSubtitle.story = {
  name: 'header subtitle component',
};
