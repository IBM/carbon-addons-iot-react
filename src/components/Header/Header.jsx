import {
  Header as CarbonHeader,
  HeaderMenuButton,
  HeaderName,
  HeaderGlobalBar,
  HeaderGlobalAction,
  SkipToContent,
  HeaderMenuItem,
  HeaderPanel,
} from 'carbon-components-react/lib/components/UIShell';
import AppSwitcher from '@carbon/icons-react/lib/app-switcher/20';
import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { settings } from 'carbon-components';
import cn from 'classnames';

import HeaderMenu from './HeaderMenu';

const { prefix: carbonPrefix } = settings;

const propTypes = {
  /** Add a prefix other than IBM */
  prefix: PropTypes.string,
  /** Name to follow the IBM prefix up top, left */
  appName: PropTypes.string.isRequired,
  /** Add a class name to Header */
  className: PropTypes.string,
  /** Provide ID for the skip to content functionality */
  skipto: PropTypes.string,
  /** href optional url to file if you click on title */
  url: PropTypes.string,
  /** Object of action items */
  actionItems: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      onClick: PropTypes.func,
      /** declare control of header panel from this action item.  */
      hasHeaderPanel: PropTypes.bool,
      btnContent: PropTypes.any.isRequired,
      childContent: PropTypes.arrayOf(
        PropTypes.shape({
          /** extra data to pass to HeaderMenuLink (aria-*, href, target, etc.) */
          metaData: PropTypes.object,
          /** Optionally pass in an onClick handler to trigger action  */
          onClick: PropTypes.func,
          content: PropTypes.any.isRequired,
        })
      ),
    })
  ).isRequired,
  /** Bit to flip that tells header to render the nav toggle button */
  hasSideNav: PropTypes.bool,
  onClickSideNavExpand: PropTypes.func,
  /** Main app switcher Header panel props */
  headerPanel: PropTypes.shape({
    /** Optionally provide a custom class to apply to the underlying <li> node */
    className: PropTypes.string,
    /** the content of the header panel  */
    content: PropTypes.any,
  }),
};

const defaultProps = {
  onClickSideNavExpand: null,
  hasSideNav: true,
  prefix: 'IBM',
  className: 'main-header',
  skipto: '#main-content',
  headerPanel: null,
  url: '#',
};

/**
 * Clickable card that shows "Add" button
 */
const Header = ({
  appName,
  className,
  actionItems,
  prefix,
  skipto,
  hasSideNav,
  onClickSideNavExpand,
  headerPanel,
  url,
}) => {
  const [expandedItem, setExpandedItem] = useState({});

  // expanded state  for headerpanels
  const handleExpandedState = index => {
    if (index) {
      setExpandedItem({
        [index]: !expandedItem[index],
      });
    } else {
      setExpandedItem(prev => {
        const oldHeaderPanel = Object.keys(prev)[0];
        return {
          [oldHeaderPanel]: false,
        };
      });
    }
  };

  // close header panel when focus is lost
  const handleCloseOnTab = e => {
    if (!e.target.parentNode.contains(e.relatedTarget)) {
      setExpandedItem(prev => {
        const oldHeaderPanel = Object.keys(prev)[0];
        return {
          [oldHeaderPanel]: false,
        };
      });
    }
  };

  const handleClickOutside = (e, label) => {
    if (
      // not header panel trigger
      e?.relatedTarget?.title !== label &&
      // not in headerpanel
      !e?.target?.parentNode?.parentNode?.parentNode.contains(e.relatedTarget)
    ) {
      // close panel
      handleExpandedState();
    }
  };

  const actionBtnHeaderPanels = [];
  const actionBtnContent = actionItems.map((item, i) => {
    if (item.hasOwnProperty('childContent')) {
      if (item.hasOwnProperty('hasHeaderPanel')) {
        const panelChildren = item.childContent.map(childItem => {
          const ChildElement = childItem?.metaData?.element || 'a';
          return (
            <li key={`listitem-${i * Math.random()}`} className="action-btn__headerpanel-li">
              <ChildElement
                key={`headerpanelmenu-item-${item.label +
                  item.childContent.indexOf(childItem)}-child-${i}`}
                {...childItem.metaData}
              >
                {childItem.content}
              </ChildElement>
            </li>
          );
        });

        return (
          <div
            data-testid="action-btn__group"
            className={`${carbonPrefix}--header__submenu ${carbonPrefix}--header-action-btn action-btn__group`}
            key={`submenu-${i}`}
          >
            <HeaderGlobalAction
              className={`${carbonPrefix}--header-action-btn action-btn__trigger`}
              key={`menu-item-${item.label}-global`}
              title={item.label}
              aria-label={item.label}
              aria-haspopup="menu"
              aria-expanded={expandedItem[item.label]}
              onClick={() => handleExpandedState(item.label)}
              onBlur={e => handleCloseOnTab(e)}
            >
              {item.btnContent}
            </HeaderGlobalAction>
            <HeaderPanel
              data-testid="action-btn__panel"
              tabIndex="-1"
              onBlur={e => handleClickOutside(e, item.label)}
              key={`panel-${i}`}
              aria-label="Header Panel"
              className={cn('action-btn__headerpanel', {
                'action-btn__headerpanel--closed': !expandedItem[item.label],
              })}
              expanded={expandedItem[item.label]}
            >
              <ul aria-label={item.label}>{panelChildren}</ul>
            </HeaderPanel>
          </div>
        );
      }
      const children = item.childContent.map(childItem => (
        <HeaderMenuItem
          key={`menu-item-${item.label + item.childContent.indexOf(childItem)}-child`}
          {...childItem.metaData}
        >
          {childItem.content}
        </HeaderMenuItem>
      ));
      return (
        <div
          data-testid="headermenu"
          onFocus={handleExpandedState}
          className={`${carbonPrefix}--header__submenu`}
          key={`wrapper-${i}`}
        >
          <HeaderMenu
            className={`${carbonPrefix}--header-action-btn`}
            key={`menu-item-${item.label}`}
            aria-label={item.label}
            isMenu={false}
            renderMenuContent={() => item.btnContent}
            menuLinkName={item.menuLinkName ? item.menuLinkName : ''}
          >
            {children}
          </HeaderMenu>
        </div>
      );
    }
    return (
      <HeaderGlobalAction
        className={`${carbonPrefix}--header-action-btn`}
        key={`menu-item-${item.label}-global-${i}`}
        aria-label={item.label}
        onClick={() => {
          handleExpandedState();
          item.onClick();
        }}
      >
        {item.btnContent}
      </HeaderGlobalAction>
    );
  });
  if (headerPanel) {
    actionBtnContent.push(
      <HeaderGlobalAction
        aria-label="header-panel-trigger"
        key="AppSwitcher"
        onClick={() => handleExpandedState('AppSwitcher')}
        title="AppSwitcher"
      >
        <AppSwitcher fill="white" description="Icon" />
      </HeaderGlobalAction>
    );
  }

  return (
    <CarbonHeader className={className} aria-label="main header">
      <SkipToContent href={skipto} />
      {hasSideNav && <HeaderMenuButton aria-label="Open menu" onClick={onClickSideNavExpand} />}
      <HeaderName href={url} prefix={prefix}>
        {appName}
      </HeaderName>
      <HeaderGlobalBar>{actionBtnContent}</HeaderGlobalBar>
      {actionBtnHeaderPanels}
      {headerPanel && (
        <HeaderPanel
          aria-label="Header Panel"
          className={cn(`${carbonPrefix}--app-switcher`, {
            [headerPanel.className]: headerPanel.className,
          })}
          expanded={expandedItem.AppSwitcher}
        >
          <div>
            <headerPanel.content />
          </div>
        </HeaderPanel>
      )}
    </CarbonHeader>
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
