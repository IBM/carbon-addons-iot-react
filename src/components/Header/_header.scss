@import '~carbon-components/scss/globals/scss/vars';

.#{$prefix}--header {
  .#{$prefix}--skip-to-content:focus {
    min-width: 240px;
    justify-content: center;
    border: $spacing-01 solid $ui-01;
  }

  &__menu-toggle {
    display: block;
  }

  &__name {
    font-size: $spacing-01;
  }

  &__menu-title[role='menuitem'][aria-expanded='true'] + &__menu {
    left: auto;
    right: 0;
  }

  &__menu {
    min-width: 12.5rem;
    width: auto;
  }

  &__menu-item[role='menuitem']:focus {
    border-color: $ui-01;
    outline: none;
  }
}

.#{$prefix}--header-action-btn {
  height: 3rem;
  width: 3rem;
  padding: 0;

  .#{$prefix}--header__menu-item[role='menuitem'] {
    height: 100%;
    justify-content: center;
    padding: 0;
    width: 100%;
  }

  span {
    display: flex;
  }
}

.#{$prefix}--header__menu .#{$prefix}--header__menu-item[role='menuitem'] {
  display: flex;
  align-items: center;
  color: #c6c6c6;
  height: 100%;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.125rem;
  text-decoration: none;
  user-select: none;
  border: 2px solid #0000;
  transition: background-color 110ms, border-color 110ms, color 110ms;
  justify-content: center;
  padding: 0;
  width: 100%;

  &:focus {
    border-color: white;
  }
}

.#{$prefix}--app-switcher {
  background-color: $ui-01;
  border: none;
}

.action-btn__headerpanel {
  outline: none;

  &--closed * {
    display: none;
  }
}
