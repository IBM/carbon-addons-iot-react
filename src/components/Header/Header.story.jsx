import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { rem } from 'polished';
import styled from 'styled-components';
import Notification from '@carbon/icons-react/lib/notification/24';
import Help from '@carbon/icons-react/lib/help/24';
import UserAvatar from '@carbon/icons-react/lib/user--avatar/24';

import Header from './Header';

React.Fragment = ({ children }) => children;

const User = styled.p`
   {
    color: white;
    font-size: 0.75rem;
    text-align: left;
    margin-right: ${rem(20)};
  }
`;

const DropItem = styled.span`
   {
    display: flex;
  }
`;

// const link = <Icon name="header--help" fill="white" description="Icon" />;
const HeaderProps = {
  user: 'JohnDoe@ibm.com',
  tenant: 'TenantId: Acme',
  className: 'custom-class-name',
  appName: 'Watson IoT Platform ',
  actionItems: [
    {
      label: 'alerts',
      onClick: action('pop alert modal'),
      btnContent: <Notification fill="white" description="Icon" />,
    },
    {
      label: 'help',
      onClick: action('help drop open'),
      btnContent: (
        <Help
          fill="white"
          description="Icon"
          className="bx--header__menu-item bx--header__menu-title"
        />
      ),
      childContent: [
        {
          onCLick: action('go to source'),
          content: <p>This is a link</p>,
        },
        {
          onCLick: action('go to source'),
          content: (
            <React.Fragment>
              <DropItem>
                JohnDoe@ibm.com
                <UserAvatar name="header--avatar" fill="white" description="Icon" />
              </DropItem>
            </React.Fragment>
          ),
        },
      ],
    },
    {
      label: 'user',
      onClick: action('click'),
      btnContent: (
        <React.Fragment>
          <User>
            JohnDoe@ibm.com<span>TenantId: Acme</span>
          </User>
          <UserAvatar name="header--avatar" fill="white" description="Icon" />
        </React.Fragment>
      ),
    },
  ],
};

storiesOf('Header', module).add('Header action buttons with dropdowns', () => (
  <Header {...HeaderProps} />
));
