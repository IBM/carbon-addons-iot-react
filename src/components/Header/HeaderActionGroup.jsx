import React from 'react';
import { HeaderGlobalBar } from 'carbon-components-react/es/components/UIShell';
import PropTypes from 'prop-types';

import HeaderAction from './HeaderAction/HeaderAction';
import { HeaderActionItemPropTypes } from './Header';

const propTypes = {
  actionItems: PropTypes.arrayOf(PropTypes.shape(HeaderActionItemPropTypes)).isRequired,
};

/**
 * Keeps track of the state of which header menu item is currently expanded
 *
 * Renders all the actions that can be clicked to navigate, open header panels (side panels),
 * or dropdown menus, passing an onToggleExpansion to each action
 */
const HeaderActionGroup = ({ actionItems }) => {
  return (
    <>
      <HeaderGlobalBar>
        {actionItems.map((item, i) => (
          <HeaderAction item={item} index={i} key={`header-action-item-${item.label}-${i}`} />
        ))}
      </HeaderGlobalBar>
    </>
  );
};

HeaderActionGroup.propTypes = propTypes;

export default HeaderActionGroup;
