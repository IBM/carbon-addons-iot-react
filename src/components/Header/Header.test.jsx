import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Notification from '@carbon/icons-react/lib/notification/20';
import Avatar from '@carbon/icons-react/lib/user--avatar/20';
import HeaderHelp from '@carbon/icons-react/lib/help/20';

import Header from './Header';

React.Fragment = ({ children }) => children;

describe('Header testcases', () => {
  const onClick = jest.fn();
  const actionItems = [
    {
      label: 'alerts',
      onClick,
      btnContent: <Notification fill="white" description="Icon" />,
    },
    {
      label: 'help',
      hasHeaderPanel: true,
      btnContent: (
        <HeaderHelp
          fill="white"
          description="Icon"
          className="bx--header__menu-item bx--header__menu-title"
        />
      ),
      childContent: [
        {
          onCLick: () => console.log('hi'),
          content: <p>This is a link</p>,
        },
        {
          onCLick: () => console.log('hi'),
          content: (
            <React.Fragment>
              <span>
                JohnDoe@ibm.com
                <Avatar fill="white" description="Icon" />
              </span>
            </React.Fragment>
          ),
        },
      ],
    },
    {
      label: 'other help',
      onClick,
      btnContent: (
        <HeaderHelp
          fill="white"
          description="Icon"
          className="bx--header__menu-item bx--header__menu-title"
        />
      ),
      childContent: [
        {
          onCLick: () => console.log('hi'),
          content: (
            <React.Fragment>
              <span>
                JohnDoe@ibm.com
                <Avatar fill="white" description="Icon" />
              </span>
            </React.Fragment>
          ),
        },
      ],
    },
  ];
  it('should render', () => {
    const { container } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('children should render inside UL', () => {
    const { getByText } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );
    // fireEvent.click(getByTitle('help'));
    // fireEvent.focus(getByText('This is a link'));
    expect(getByText('This is a link').parentElement.parentElement.nodeName).toBe('LI');
    expect(getByText('This is a link').parentElement.parentElement.className).toEqual(
      'action-btn__headerpanel-li'
    );
    expect(
      getByText('This is a link').parentElement.parentElement.parentNode.childElementCount
    ).toEqual(2);
    // fireEvent.blur(container.querySelector('.action-btn__group'));
  });

  it('clicking trigger button should expand the header panel', () => {
    const { getByTitle } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );
    fireEvent.click(getByTitle('help'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel bx--header-panel--expanded action-btn__headerpanel'
    );
    fireEvent.click(getByTitle('help'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel action-btn__headerpanel action-btn__headerpanel--closed'
    );
  });

  it('closes when focus leaves panel', () => {
    const { getByTestId, getByTitle } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );
    fireEvent.click(getByTitle('help'));
    fireEvent.focus(getByTestId('action-btn__panel'));
    fireEvent.blur(getByTestId('action-btn__panel'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel action-btn__headerpanel action-btn__headerpanel--closed'
    );
  });

  it('closes when focus leaves trigger', () => {
    const { getByTitle } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );
    fireEvent.click(getByTitle('help'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel bx--header-panel--expanded action-btn__headerpanel'
    );
    fireEvent.blur(getByTitle('help'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel action-btn__headerpanel action-btn__headerpanel--closed'
    );
  });

  it('closes when focus goes to next action btn', () => {
    const { getByTitle, getByTestId } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
      />
    );

    fireEvent.click(getByTitle('help'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel bx--header-panel--expanded action-btn__headerpanel'
    );
    fireEvent.focus(getByTestId('headermenu'));
    expect(getByTitle('help').parentNode.lastChild.className).toContain(
      'bx--header-panel action-btn__headerpanel action-btn__headerpanel--closed'
    );
  });

  it('App switcher opens', () => {
    const headerPanel = {
      className: 'header-panel',
      /* eslint-disable */

      content: React.forwardRef((props, ref) => (
        <a href="#" ref={ref} {...props}>
          Header panel content
        </a>
      )),
      /* eslint-enable */
    };
    const { getByTitle } = render(
      <Header
        title="My Title"
        user="j@test.com"
        tenant="acme"
        appName="platform"
        actionItems={actionItems}
        headerPanel={headerPanel}
      />
    );

    fireEvent.click(getByTitle('AppSwitcher'));
    expect(getByTitle('AppSwitcher').parentNode.nextSibling.className).toContain(
      'bx--header-panel bx--header-panel--expanded bx--app-switcher header-panel'
    );
    fireEvent.click(getByTitle('AppSwitcher'));
    expect(getByTitle('AppSwitcher').parentNode.nextSibling.className).toContain(
      'bx--header-panel bx--app-switcher header-panel'
    );
  });
});
