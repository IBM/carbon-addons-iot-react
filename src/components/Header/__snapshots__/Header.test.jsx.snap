// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testcases should render 1`] = `
<div>
  <header
    aria-label="main header"
    class="bx--header main-header"
    role="banner"
  >
    <a
      class="bx--skip-to-content"
      href="#main-content"
      tabindex="0"
    >
      Skip to main content
    </a>
    <button
      aria-label="Open menu"
      class="bx--header__action bx--header__menu-trigger bx--header__menu-toggle bx--header__menu-toggle__hidden"
      title="Open menu"
      type="button"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 14.8h16V16H2zm0-3.6h16v1.2H2zm0-3.6h16v1.2H2zM2 4h16v1.2H2z"
        />
      </svg>
    </button>
    <a
      class="bx--header__name"
      href="#"
    >
      <span
        class="bx--header__name--prefix"
      >
        IBM
      </span>
      Â 
      platform
    </a>
    <div
      class="bx--header__global"
    >
      <button
        aria-label="alerts"
        class="bx--header-action-btn bx--header__action"
        type="button"
      >
        <svg
          aria-hidden="true"
          description="Icon"
          fill="white"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M28.7 20.3L26 17.6V13a10.07 10.07 0 0 0-9-10V1h-2v2a10.15 10.15 0 0 0-9 10v4.6l-2.7 2.7a.91.91 0 0 0-.3.7v3a.94.94 0 0 0 1 1h7a5 5 0 0 0 10 0h7a.94.94 0 0 0 1-1v-3a.91.91 0 0 0-.3-.7zM16 28a3 3 0 0 1-3-3h6a3 3 0 0 1-3 3zm11-5H5v-1.6l2.7-2.7A.91.91 0 0 0 8 18v-5a8 8 0 0 1 16 0v5a.91.91 0 0 0 .3.7l2.7 2.7z"
          />
        </svg>
      </button>
      <div
        class="bx--header__submenu bx--header-action-btn action-btn__group"
        data-testid="action-btn__group"
      >
        <button
          aria-haspopup="menu"
          aria-label="help"
          class="bx--header-action-btn action-btn__trigger bx--header__action"
          title="help"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--header__menu-item bx--header__menu-title"
            description="Icon"
            fill="white"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 2a14 14 0 1 0 14 14A14 14 0 0 0 16 2zm0 26a12 12 0 1 1 12-12 12 12 0 0 1-12 12z"
            />
            <circle
              cx="16"
              cy="23.5"
              r="1.5"
            />
            <path
              d="M17 8h-1.5a4.49 4.49 0 0 0-4.5 4.5v.5h2v-.5a2.5 2.5 0 0 1 2.5-2.5H17a2.5 2.5 0 0 1 0 5h-2v4.5h2V17a4.5 4.5 0 0 0 0-9z"
            />
          </svg>
        </button>
        <div
          aria-label="Header Panel"
          class="bx--header-panel action-btn__headerpanel action-btn__headerpanel--closed"
          data-testid="action-btn__panel"
          tabindex="-1"
        >
          <ul
            aria-label="help"
          >
            <li
              class="action-btn__headerpanel-li"
            >
              <a>
                <p>
                  This is a link
                </p>
              </a>
            </li>
            <li
              class="action-btn__headerpanel-li"
            >
              <a>
                <span>
                  JohnDoe@ibm.com
                  <svg
                    aria-hidden="true"
                    description="Icon"
                    fill="white"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 2a14 14 0 1 0 14 14A14 14 0 0 0 16 2zm-6 24.38v-2A3.22 3.22 0 0 1 13 21h6a3.22 3.22 0 0 1 3 3.39v2a11.92 11.92 0 0 1-12 0zm14-1.46v-.61A5.21 5.21 0 0 0 19 19h-6a5.2 5.2 0 0 0-5 5.31v.59a12 12 0 1 1 16 0z"
                    />
                    <path
                      d="M16 7a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
                    />
                  </svg>
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="bx--header__submenu"
        data-testid="headermenu"
      >
        <div
          class="bx--header__submenu bx--header-action-btn"
        >
           
          <a
            aria-expanded="false"
            aria-haspopup="menu"
            aria-label="other help"
            class="bx--header__menu-item bx--header__menu-title"
            data-testid="menuitem"
            href="javascript:void(0)"
            role="menuitem"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="bx--header__menu-item bx--header__menu-title"
              description="Icon"
              fill="white"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 2a14 14 0 1 0 14 14A14 14 0 0 0 16 2zm0 26a12 12 0 1 1 12-12 12 12 0 0 1-12 12z"
              />
              <circle
                cx="16"
                cy="23.5"
                r="1.5"
              />
              <path
                d="M17 8h-1.5a4.49 4.49 0 0 0-4.5 4.5v.5h2v-.5a2.5 2.5 0 0 1 2.5-2.5H17a2.5 2.5 0 0 1 0 5h-2v4.5h2V17a4.5 4.5 0 0 0 0-9z"
              />
            </svg>
          </a>
          <ul
            aria-label="other help"
            class="bx--header__menu"
            role="menu"
          >
            <li
              role="none"
            >
              <a
                class="bx--header__menu-item"
                role="menuitem"
                tabindex="0"
              >
                <span
                  class="bx--text-truncate--end"
                >
                  <span>
                    JohnDoe@ibm.com
                    <svg
                      aria-hidden="true"
                      description="Icon"
                      fill="white"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 2a14 14 0 1 0 14 14A14 14 0 0 0 16 2zm-6 24.38v-2A3.22 3.22 0 0 1 13 21h6a3.22 3.22 0 0 1 3 3.39v2a11.92 11.92 0 0 1-12 0zm14-1.46v-.61A5.21 5.21 0 0 0 19 19h-6a5.2 5.2 0 0 0-5 5.31v.59a12 12 0 1 1 16 0z"
                      />
                      <path
                        d="M16 7a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </li>
          </ul>
           
        </div>
      </div>
    </div>
  </header>
</div>
`;
